iFixit/checkout-ui-extensions
# Add newsletter subscription

closes https://github.com/iFixit/checkout-ui-extensions/issues/17
~~awating https://github.com/iFixit/ifixit/pull/49782 to merge~~

Summary
--
We want to include a newsletter subscription section to `Thank you` and `Order Status` pages.

![image](https://github.com/iFixit/checkout-ui-extensions/assets/95708883/167b704f-bb94-41dc-ba5d-c1fd0ffdcb31)

QA
--

1. Check out this branch, run `pnpm i`, then `pnpm dev`
2. Open the preview console, then click on `purchase.thank-you.block.render`
3. Append `?placement-reference=ORDER_STATUS3` to the url
4. Complete a checkout with any item, and once you land on the `Thank you` page (after completing purchase) confirm you can see the component
5. Test that you are able to subscribe to the newsletter (can check network calls to verify this)
6. Refresh the `thank you` page that you are currently on. Observe the url changes to show the `order status` page. If the url doesn't change, try appending the `?prevent_order_redirect=false` param to the url. If that doesn't work, feel free to skip this step and the next step and leave a comment saying that this was skipped.
7. Observe that the component does appear and the functionality is as described before
